### DISCUSSION TITLE
Update mcap file with more detailed schemas?

### DISCUSSION THREAD

**Discussion by freol35241** [foxglove/mcap#1122] [General] [answered]
Currently we (ab-)use the possibility of writing JSON using a schema that accepts everything in instances where we don't know about the schema beforehand.

In those situations it would be handy to be able to update the schemas of specific channels on existing files once the schema is known. I have not found any way of doing this so I guess it is not supported without rewritinh the whole file?

**Comment by jtbandes** [comment, marked as answer]
This is not supported by the MCAP format – a schema or channel cannot be changed. One way of addressing this would be to avoid writing messages until the schema is known (e.g. holding them in memory up to some limit).

**Reply by freol35241** [reply]
Thanks for the answer. Holding messages in memory is not a viable option for us. Will resort to rewriting the file with new schemas where required.
